packages/typespec-client-generator-core/design-docs/client.md

```diff
@@ -0,0 +1,564 @@
+# TCGC client related type design
+
+We have many client related types, decorators and concepts in TCGC and some of them are implicit, which makes author hard to know how to generate a needed client. This doc aims to revisit all client related things in TCGC and try to have a consolidated design output.
+
+## User scenario
+
+1. Single client
+
+```Python
```

Comment:
It would be great to add one strongly-typed language sample as well in the doc.

Reply:
i think this is up to separate languages to add